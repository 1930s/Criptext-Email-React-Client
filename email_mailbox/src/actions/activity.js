import { Activity } from './types';

export const startLoadSync = () => {
  return {
    type: Activity.START_LOAD_SYNC
  };
};

export const stopLoadSync = () => {
  return {
    type: Activity.STOP_LOAD_SYNC
  };
};

export const startLoadThread = () => {
  return {
    type: Activity.START_LOAD_THREAD
  };
};

export const stopLoadThread = () => {
  return {
    type: Activity.STOP_LOAD_THREAD
  };
};
