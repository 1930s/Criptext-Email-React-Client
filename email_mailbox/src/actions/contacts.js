import { Contact } from './types';
import * as db from '../utils/electronInterface';

export const addContacts = contacts => {
  return {
    type: Contact.ADD_BATCH,
    contacts: contacts
  };
};

export const loadContacts = ids => {
  return async dispatch => {
    try {
      const response = await db.getContactByIds(ids);
      const contacts = {};
      response.forEach(element => {
        contacts[element.id] = element;
      });
      dispatch(addContacts(contacts));
    } catch (e) {
      // TO DO
    }
  };
};
