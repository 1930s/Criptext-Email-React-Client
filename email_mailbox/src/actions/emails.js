import { Email } from './types';
import {
  getEmailsByThreadId,
  setMuteEmailById
} from '../utils/electronInterface';

export const addEmails = emails => {
  return {
    type: Email.ADD_BATCH,
    emails: emails
  };
};

export const muteNotifications = emailId => {
  return {
    type: Email.MUTE,
    targetEmail: emailId
  };
};

export const loadEmails = threadId => {
  return async dispatch => {
    try {
      const response = await getEmailsByThreadId(threadId);
      const emails = {};
      response.forEach(element => {
        emails[element.id] = element;
      });
      dispatch(addEmails(emails));
    } catch (e) {
      // TO DO
    }
  };
};

export const muteEmailById = feed => {
  return async dispatch => {
    try {
      const valueToSet = feed.get('isMuted') === 1 ? false : true;
      await setMuteEmailById(feed.get('emailId'), valueToSet);
      dispatch(muteNotifications(feed.get('emailId')));
    } catch (e) {
      // To do
    }
  };
};
