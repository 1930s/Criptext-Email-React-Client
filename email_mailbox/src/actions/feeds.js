import { Feed, Thread } from './types';

export const addFeeds = feeds => {
  return {
    type: Feed.ADD_BATCH,
    feeds: feeds
  };
};

export const muteNotifications = threadId => {
  return {
    type: Thread.MUTE_THREAD,
    targetThread: threadId
  };
};

export const toggleMuteFeed = feedId => {
  return {
    type: Feed.TOGGLE_MUTE,
    targetFeed: feedId
  };
};

export const removeFeed = feedId => {
  return {
    type: Feed.REMOVE,
    targetFeed: feedId
  };
};

export const selectFeed = feedId => {
  return {
    type: Feed.SELECT,
    selectedFeed: feedId
  };
};

export const loadFeeds = () => {
  return async dispatch => {
    try {
      const response = await fetch('/feeds.json');
      const json = await response.json();
      const feeds = json.feeds;
      dispatch(addFeeds(feeds));
    } catch (e) {
      // TO DO
    }
  };
};
