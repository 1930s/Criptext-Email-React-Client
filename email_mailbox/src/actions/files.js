import { File } from './types';
import * as db from '../utils/electronInterface';

export const addFiles = files => {
  return {
    type: File.ADD_BATCH,
    files
  };
};

export const loadFiles = tokens => {
  return async dispatch => {
    try {
      const response = await db.getFilesByTokens(tokens);
      const files = response.reduce(
        (result, file) => ({
          ...result,
          [file.token]: file
        }),
        {}
      );
      dispatch(addFiles(files));
    } catch (e) {
      // TO DO
    }
  };
};
