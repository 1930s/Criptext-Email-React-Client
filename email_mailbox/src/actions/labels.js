import { Label } from './types';
import { getAllLabels } from '../utils/electronInterface';
const inLabel = 10;

export const addLabels = labels => {
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const addLabel = label => {
  const labels = {
    inLabel: {
      id: inLabel,
      text: label
    }
  };
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const loadLabels = () => {
  return async dispatch => {
    try {
      const response = await getAllLabels();
      const labels = {};
      response.forEach(element => {
        labels[element.id] = {
          id: element.id,
          color: element.color,
          text: element.text
        };
      });
      dispatch(addLabels(labels));
    } catch (e) {
      // TO DO
    }
  };
};
