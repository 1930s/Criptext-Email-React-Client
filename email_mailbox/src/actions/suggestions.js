import { Suggestions } from './types';
import { getEmailsGroupByThreadByMatchText } from '../utils/electronInterface';
import { getMatches } from '../utils/storage';

export const setSuggestions = (threads, hints) => ({
  type: Suggestions.SET_THREADS,
  threads,
  hints
});

export const setSuggestionError = error => ({
  type: Suggestions.SET_ERROR_SUGGESTIONS,
  error: error
});

export const loadSuggestions = filter => {
  return async dispatch => {
    try {
      const hints = await getMatches(filter);
      const threads = await getEmailsGroupByThreadByMatchText(filter);
      dispatch(setSuggestions(threads, hints));
    } catch (e) {
      dispatch(setSuggestionError('Unable to load suggestions'));
    }
  };
};
