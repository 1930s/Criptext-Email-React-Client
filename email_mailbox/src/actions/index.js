import { addUsers, addUser } from './users';
import {
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  filterThreadsByUnread,
  loadThreads,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  selectThread,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeThreads,
  removeThreadsLabel
} from './threads';
import { addEmails, loadEmails } from './emails';
import { addLabels, addLabel, loadLabels } from './labels';
import {
  addFeeds,
  loadFeeds,
  removeFeed,
  selectFeed,
  toggleMuteFeed
} from './feeds';
import { setThreads, loadThreadsSuggestions } from './threadsSuggestions'

export {
  addEmails,
  addFeeds,
  addLabel,
  addLabels,
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  addUser,
  addUsers,
  filterThreadsByUnread,
  loadEmails,
  loadFeeds,
  loadLabels,
  loadThreads,
  loadThreadsSuggestions,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeFeed,
  removeThreads,
  removeThreadsLabel,
  selectThread,
  selectFeed,
  setThreads,
  toggleMuteFeed
};
