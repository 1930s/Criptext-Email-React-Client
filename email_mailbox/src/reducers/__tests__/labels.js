/* eslint-env node, jest */

import labelReducer from './../labels';
import * as actions from '../../actions/index';

jest.mock('../../utils/electronInterface');

describe('label actions: ', () => {
  it('should add labels to state', () => {
    const labels = {
      '1': {
        id: 1,
        text: 'label 1'
      },
      '2': {
        id: 2,
        text: 'label 2'
      }
    };
    const action = actions.addLabels(labels);
    const state = labelReducer(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
