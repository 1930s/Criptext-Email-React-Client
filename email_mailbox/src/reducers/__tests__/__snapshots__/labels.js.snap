// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label actions:: should add labels to state 1`] = `
Immutable.Map {
  1: Immutable.Map {
    id: 1,
    color: "#ffffff",
    text: "Inbox",
  },
  2: Immutable.Map {
    id: 2,
    color: "#ffffff",
    text: "Sent",
  },
  3: Immutable.Map {
    id: 3,
    color: "#ffffff",
    text: "Spam",
  },
  4: Immutable.Map {
    id: 4,
    color: "#ffffff",
    text: "Trash",
  },
  5: Immutable.Map {
    id: 5,
    color: "#ffffff",
    text: "Draft",
  },
  6: Immutable.Map {
    id: 6,
    color: "#ffffff",
    text: "Important",
  },
  7: Immutable.Map {
    id: 7,
    color: "#ffffff",
    text: "Starred",
  },
  10: Immutable.Map {
    id: 10,
    color: "#ffffff",
    text: "Text 1",
  },
}
`;

exports[`Label actions:: should modify label: color 1`] = `
Immutable.Map {
  1: Immutable.Map {
    id: 1,
    color: "#000000",
    text: "Inbox",
  },
  2: Immutable.Map {
    id: 2,
    color: "#ffffff",
    text: "Sent",
  },
  3: Immutable.Map {
    id: 3,
    color: "#ffffff",
    text: "Spam",
  },
  4: Immutable.Map {
    id: 4,
    color: "#ffffff",
    text: "Trash",
  },
  5: Immutable.Map {
    id: 5,
    color: "#ffffff",
    text: "Draft",
  },
  6: Immutable.Map {
    id: 6,
    color: "#ffffff",
    text: "Important",
  },
  7: Immutable.Map {
    id: 7,
    color: "#ffffff",
    text: "Starred",
  },
  10: Immutable.Map {
    id: 10,
    color: "#ffffff",
    text: "Text 1",
  },
}
`;

exports[`Label actions:: should modify label: text 1`] = `
Immutable.Map {
  1: Immutable.Map {
    id: 1,
    color: "#ffffff",
    text: "labelmodified",
  },
  2: Immutable.Map {
    id: 2,
    color: "#ffffff",
    text: "Sent",
  },
  3: Immutable.Map {
    id: 3,
    color: "#ffffff",
    text: "Spam",
  },
  4: Immutable.Map {
    id: 4,
    color: "#ffffff",
    text: "Trash",
  },
  5: Immutable.Map {
    id: 5,
    color: "#ffffff",
    text: "Draft",
  },
  6: Immutable.Map {
    id: 6,
    color: "#ffffff",
    text: "Important",
  },
  7: Immutable.Map {
    id: 7,
    color: "#ffffff",
    text: "Starred",
  },
  10: Immutable.Map {
    id: 10,
    color: "#ffffff",
    text: "Text 1",
  },
}
`;

exports[`Label actions:: should modify label: text and color 1`] = `
Immutable.Map {
  1: Immutable.Map {
    id: 1,
    color: "#000000",
    text: "labelmodified",
  },
  2: Immutable.Map {
    id: 2,
    color: "#ffffff",
    text: "Sent",
  },
  3: Immutable.Map {
    id: 3,
    color: "#ffffff",
    text: "Spam",
  },
  4: Immutable.Map {
    id: 4,
    color: "#ffffff",
    text: "Trash",
  },
  5: Immutable.Map {
    id: 5,
    color: "#ffffff",
    text: "Draft",
  },
  6: Immutable.Map {
    id: 6,
    color: "#ffffff",
    text: "Important",
  },
  7: Immutable.Map {
    id: 7,
    color: "#ffffff",
    text: "Starred",
  },
  10: Immutable.Map {
    id: 10,
    color: "#ffffff",
    text: "Text 1",
  },
}
`;
